src/main/java/core/characters/CharacterManager.java

```diff
@@ -1,80 +1,145 @@
+                this.keylist.add(maxElement + 1);
+            } else {
+                this.characterEntities.put(maxElement + 2, character);
+                this.keylist.add(maxElement + 2);
+            }
+        }
     }
 
-    public void moveCharacter() {
-        // Updates the characters position
+    public void moveCharacter(int id, float newX, float newY) {
+        //Updates the position of the character
+        for (var i : this.characterEntities.entrySet()) {
+            if (i.getKey() == id) {
+                i.getValue().position[0] = newX;
+                i.getValue().position[1] = newY;
+            }
+        }
     }
 
-    public void depleteHealth(int damage) {
+    public void depleteHealth(int id, int damage) {
         /*
-         * Decreases player health by damage
+         * Decreases character health by damage
          * */
-        this.character.health -= damage;
+        for (var i : this.characterEntities.entrySet()) {
+            if (i.getKey() == id) {
+                i.getValue().health -= damage;
+            }
+        }
     }
 
-    public void increaseLevel() {
+    public void increaseLevel(int id) {
         /*
          * Increases the level of enemies by 1 following each wave of enemies.
          * Need to check if character.team == "Enemy"
          * */
-        this.character.level += 1;
+        for (var i : this.characterEntities.entrySet()) {
+            if (i.getKey() == id) {
+                i.getValue().level += 1;
+            }
+        }
     }
 
-    public boolean useItem() {
+    public boolean canUseItem(int id, String item) {
         /*
-         * Checks if the item is in the characters inventory and then allows them to use it.
-         * This will only be available to the player class, will check insanceof Player.
-         * Returns true iff the player uses the item, false otherwise.
+         * Checks if the item is in the characters inventory and then returns true if it is.
+         * Ensures that there are no issues when controller class calls a child of itemUsageDelegate
          * */
 
-        return true;
+        for (var i : this.characterEntities.entrySet()) {
+            if (i.getKey() == id) {
+                if (i.getValue().inventory.contains(item)) {
+                    return true;
+                }
+            }
+        }
+        return false;
     }
 
-    public void addInventory(String item) {
+    public void addInventory(int id, String item) {
         /*
          * Adds item to the inventory
-         * Only valid for the player class, will check insanceof Player.
          * */
+        for (var i : this.characterEntities.entrySet()) {
+            if (i.getKey() == id) {
+                i.getValue().inventory.add(item);
+            }
+        }
     }
 
-    public boolean removeInventory(String item) {
+    public boolean removeInventory(int id, String item) {
```

Comment:
maybe delegate all this stuff to a separate InventoryManager – how inventory works is implementation details not super relevant to how the character works